File: C++/389.找不同.cpp
/*
 * @lc app=leetcode.cn id=389 lang=cpp
 *
 * [389] 找不同
 */

// @lc code=start
class Solution
{
public:
    char findTheDifference(string s, string t)
    {
        vector<int> cnt(26, 0);
        for (char c : s)
        {
            cnt[c - 'a']++;
        }
        for (char c : t)
        {
            cnt[c - 'a']--;
            if (cnt[c - 'a'] < 0)
            {
                return c;
            }
        }
        return ' ';
    }
};
// @lc code=end

File: JavaScript/389.找不同.js
/*
 * @lc app=leetcode.cn id=389 lang=javascript
 *
 * [389] 找不同
 */

// @lc code=start
/**
 * @param {string} s
 * @param {string} t
 * @return {character}
 */
var findTheDifference = function (s, t) {
  const count = new Map();
  s.split("").forEach((c) => {
    count.set(c, (count.get(c) || 0) + 1);
  });
  t.split("").forEach((c) => {
    count.set(c, (count.get(c) || 0) - 1);
  });
  for (const [key, val] of count) {
    if (val !== 0) {
      return key;
    }
  }
};
// @lc code=end
